docs/control-structures.rst

```diff
@@ -273,18 +273,12 @@ i.e. the following is not valid: ``(x, uint y) = (1, 2);``
 Complications for Arrays and Structs
 ------------------------------------
 
-The semantics of assignments are a bit more complicated for
-non-value types like arrays and structs.
-Assigning *to* a state variable always creates an independent
-copy. On the other hand, assigning to a local variable creates
-an independent copy only for elementary types, i.e. static
-types that fit into 32 bytes. If structs or arrays (including
-``bytes`` and ``string``) are assigned from a state variable
-to a local variable, the local variable holds a reference to
-the original state variable. A second assignment to the local
-variable does not modify the state but only changes the
-reference. Assignments to members (or elements) of the local
-variable *do* change the state.
+The semantics of assignments are more complicated for non-value types like arrays and structs,
+including ``bytes`` and ``string``:
+
+.. include:: types/reference-types.rst
```

Comment:
Hm... is there a reason against just linking here (i.e. "See [link]") instead of actually duplicating?

Reply:
@ekpyron duplicating what?

Reply:
I'm not sure it makes sense to keep this example here since it's not explained here anymore how it works.

Reply:
The contents of ``types/reference-types.rst``... at least to me it seemed like the ``include`` here actually copies the content over, but I haven't actually checked...

Reply:
We should certainly link.

Reply:
@ekpyron It intentionally copied (or to be precise reused) the content to give people what they need in one place, always best to not have to send people back and forth between documents, and with this option you only need to update the content in one place.

And the example is still explained @leonardoalt  you might need to render the docs with Sphinx to see how everything looks in place. It seems pretty nice to me, but maybe I'm missing something.

![array-copy](https://user-images.githubusercontent.com/42080/74424939-264c2d00-4e53-11ea-98ff-1400e3ea60ef.png)